components/brave_new_tab_ui/components/default/braveToday/useUnpaddedImageUrl.ts

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+import getBraveNewsController from '../../../api/brave_news'
+
+interface Options {
+  rootElement?: HTMLElement | null
+  rootMargin?: string
+  threshold?: number
+
+  useCache?: boolean
+  onLoaded?: () => any
+}
+
+const cache: { [url: string]: string } = {}
+export function useUnpaddedImageUrl (paddedUrl: string | undefined, onLoaded?: () => any, useCache?: boolean) {
+  const [unpaddedUrl, setUnpaddedUrl] = React.useState('')
+
+  React.useEffect(() => {
+    const onReceiveUnpaddedUrl = (result: string) => {
+      if (useCache) cache[paddedUrl!] = result
+      setUnpaddedUrl(result)
+
+      if (onLoaded) window.requestAnimationFrame(() => onLoaded())
+    }
+
+    // Storybook method
+    // @ts-expect-error
+    if (window.braveStorybookUnpadUrl) {
+      // @ts-expect-error
+      window.braveStorybookUnpadUrl(paddedUrl)
+        .then(onReceiveUnpaddedUrl)
+      return
+    }
+
+    if (!paddedUrl) return
```

Comment:
Can we move this up to the first line?

Reply:
I don't think so, unfortunately - the storybook override needs to work even if there isn't a url passed in (i.e. in the case of our mock data).